Build failure: blackbox-terminal

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *blackbox-terminal*

### Build log

```
[7/215] Generating src/box-drawing with a custom command (wrapped by meson to capture output)
FAILED: src/box_drawing.h 
/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/bin/meson --internal exe --capture src/box_drawing.h -- /build/source/src/box_drawing_generate.sh ../src/box_drawing.txt
Traceback (most recent call last):
  File "/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/bin/meson", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/lib/python3.11/site-packages/mesonbuild/mesonmain.py", line 286, in main
    return run(sys.argv[1:], launcher)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/lib/python3.11/site-packages/mesonbuild/mesonmain.py", line 274, in run
    return run_script_command(args[1], args[2:])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/lib/python3.11/site-packages/mesonbuild/mesonmain.py", line 215, in run_script_command
    return module.run(script_args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/lib/python3.11/site-packages/mesonbuild/scripts/meson_exe.py", line 112, in run
    return run_exe(exe)
           ^^^^^^^^^^^^
  File "/nix/store/ykjhz2wqms7fb6jnwdj9282qibm682d7-meson-1.4.0/lib/python3.11/site-packages/mesonbuild/scripts/meson_exe.py", line 56, in run_exe
    p = subprocess.Popen(cmd_args, env=child_env, cwd=exe.workdir,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/lpi16513bai8kg2bd841745vzk72475x-python3-3.11.9/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/lpi16513bai8kg2bd841745vzk72475x-python3-3.11.9/lib/python3.11/subprocess.py", line 1840, in _execute_child
    self._posix_spawn(args, executable, env, restore_signals,
  File "/nix/store/lpi16513bai8kg2bd841745vzk72475x-python3-3.11.9/lib/python3.11/subprocess.py", line 1784, in _posix_spawn
    self.pid = os.posix_spawn(executable, args, env, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/build/source/src/box_drawing_generate.sh'
[8/215] Generating src/vte/vtetypebuiltins-gtk4.cc with a custom command
[9/215] Compiling C++ object src/decoder-cat.p/utf8.cc.o
[10/215] Generating glib marshaller source src/marshal_c
INFO: Reading ../src/marshal.list...
[11/215] Generating src/modes with a custom command
[12/215] Compiling C++ object src/decoder-cat.p/debug.cc.o
[13/215] Generating src/parser-seq with a custom command
[14/215] Compiling C++ object src/decoder-cat.p/icu-decoder.cc.o
[15/215] Compiling C++ object src/decoder-cat.p/icu-converter.cc.o
[16/215] Compiling C++ object src/decoder-cat.p/glib-glue.cc.o
[17/215] Compiling C++ object src/decoder-cat.p/decoder-cat.cc.o
[18/215] Compiling C++ object src/decoder-cat.p/icu-glue.cc.o
ninja: build stopped: subcommand failed.
```

### Additional context

blackbox uses vte with [src override](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/terminal-emulators/blackbox-terminal/default.nix#L67-L73), this change probably broke it https://github.com/NixOS/nixpkgs/pull/308957/files#diff-a05b75d8e0fa8493b19cf11cad645133f4c8eefcee7eec8d9055b6d90f89a931L104

### Notify maintainers

@chuangzhu @linsui @bobby285271

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.8-xanmod1, NixOS, 24.05 (Uakari), 24.05.20240503.6367918`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/vl8llv45x1dk5fwdnm9fjqxmccrm1nsx-source`
```
but that's irrelevant as i'm trying to build the latest master.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/terminal-emulators/blackbox-terminal/default.nix
+++ b/pkgs/applications/terminal-emulators/blackbox-terminal/default.nix
@@ -72,6 +72,9 @@ stdenv.mkDerivation rec {
         rev = "3c8f66be867aca6656e4109ce880b6ea7431b895";
         hash = "sha256-vz9ircmPy2Q4fxNnjurkgJtuTSS49rBq/m61p1B43eU=";
       };
+      postPatch = (old.postPatch or "") + ''
+        patchShebangs src/box_drawing_generate.sh
+      '';
     } // lib.optionalAttrs sixelSupport {
       buildInputs = old.buildInputs ++ [ libsixel ];
       mesonFlags = old.mesonFlags ++ [ "-Dsixel=true" ];

--- a/pkgs/development/libraries/vte/default.nix
+++ b/pkgs/development/libraries/vte/default.nix
@@ -27,6 +27,7 @@
 , systemd
 , systemdSupport ? lib.meta.availableOn stdenv.hostPlatform systemd
 , nixosTests
+, blackbox-terminal
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -117,6 +118,7 @@ stdenv.mkDerivation (finalAttrs: {
     };
     tests = {
       inherit (nixosTests.terminal-emulators) gnome-terminal lxterminal mlterm roxterm sakura stupidterm terminator termite xfce4-terminal;
+      blackbox-terminal = blackbox-terminal.override { sixelSupport = true; };
     };
   };
 

